float32 speed
uint8 direction 
---
bool result
